components/src/dynamo/common/utils/image_utils.py

```
@@ -0,0 +1,51 @@
+        PNG-encoded bytes.
+
+    Raises:
+        ImportError: If Pillow is not available.
+        RuntimeError: If encoding fails.
+    """
+    try:
+        from PIL import Image
+    except ImportError:
+        raise ImportError(
+            "Pillow is required for PNG encoding. " "Install with: pip install Pillow"
+        )
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)